EXT_BeamPhysics.md

```diff
@@ -0,0 +1,264 @@
+  - Type: Optional *(string)*
+  - Description: Reference total (kinetic + rest mass) energy. Possibly used for normalizing particle momentum values.
+
+  - `speciesType`
+    - Type: Required *(string)*
+    - Description: The name of the particle species. Species names must conform to the
+    `speciesType` extension.
+    - Example: `electron`, `H2O`.
+
+
+Per-Particle Records of the `Particle Root Group`
+-------------------------------------------------
+
+The following records store data on a particle-by-particle basis.
+
+- `electricField/`
+    - Type: Optional 2-component *(float)*
+    - Description: Electric field. Used for photons only.
+    - Components: (`x`, `y`).
+        - For each component, the field is specified using either (`amplitude`, `phase`) or (`Real`, `Imaginary`)
+        subcomponents.
+
+- `latticeElementName`
+  - Type: Optional *(string)*
+  - Description: The name of the lattice element the particle or particles are in. This only makes sense if all
+  particles are in the same lattice element. Also see: `latticeElementID` and `locationInElement`.
+
+- `latticeElementID`
+  - Type Optional *(string)*
+  - Description: The ID string for the lattice element given by `latticeElementName`. The idea is that while more than
+    one lattice element may have the same name, the ID string will be unique.
+  - This, along with `locationInElement` sets the origin for specifying particle positions in the **lattice** coordinate system.
+  - Example: With [Bmad](https://www.classe.cornell.edu/bmad/) based programs the ID string is of the form
+    **branch-index>>element-index** where **branch-index** is the associated
+branch index integer, and **element-index** is the associated lattice element index within the branch.
+
+- `locationInElement`
+  - Type Optional *(integer)*
+  - Description: The program generating the data file may model a lattice element using a "hard edge" model where the fringe fields at the ends of the element are modeled as having zero longitudinal length. In such a case, if a particle is at the end of the lattice element, it is important to know if the particle is outside of the fringe or if the particle is inside the fringe within the body of the element. Note that with a hard edge fringe, the longitudinal **s**-position does not necessarily provide enough information to determine where a particle is with respect to an edge field. Another situation where `locationInElement` is useful is with zero length elements that affect the particle transport (such as zero length multipole elements). If the program generating the data file does **not** use any hard edge models or zero length non-marker elements, `locationInElement` should not be present since this parameter is meaningless in this case.
+  - Possible values:    
+    - -1: Upstream end of element outside of the upstream fringe edge.
+    - 0: Inside the element.
+    - 1: Downstream end of the element outside outside the downstream fringe edge.
```

Comment:
There is a duplication of `outside` in the above text.

Reply:
Fixed